src/org/labkey/test/components/ui/grids/QueryGrid.java

```diff
@@ -175,10 +175,15 @@ public void doAndWaitForUpdate(Runnable func)
 
     /**
      * searches the grid, from the GridBar search input and waits for the grid to refresh
+     * if the searchTerm is different from the current searchTerm
      */
     public QueryGrid search(String searchTerm)
     {
-        doAndWaitForUpdate(()-> getGridBar().searchFor(searchTerm));
+        String currentSearchExp = getGridBar().getSearchExpression();
+        if (searchTerm.equals(currentSearchExp))
+            getGridBar().searchFor(searchTerm);
+        else
+            doAndWaitForUpdate(()-> getGridBar().searchFor(searchTerm));
```

Comment:
Why not just no-op if there's nothing to do?
```suggestion
        if (!searchTerm.equals(currentSearchExp))
            doAndWaitForUpdate(()-> getGridBar().searchFor(searchTerm));
```